Refactor - Rationalize naming of MARC test data files

tfmorris opened the pull request:
No issue. This refactors the MARC unit test suite so that files have their correct extensions (and in the cases of those missing extensions, adds extensions).

This gives all JSON files a `.json` extension instead of the misleading `.mrc` and `.xml` extensions, and gives all binary MARC files a `.mrc` extension to make them easier to use with standard tools like MARCedit.

Please merge **after** #7409, #7407, and #7405 to avoid merge conflicts.


### Stakeholders
@hornc

hornc commented:
@tfmorris I'm not sure why this has merge conflicts now? Is this something you can fix quickly?

#7405 and #7407 are already merged; I'd like to get this in before #7409 if that's okay. #7409 feels like a partial feature, and I'm not sure how to compete it, but this renaming should just be merged ASAP as a refactor and clear improvement.

tfmorris commented:
I've rebased (and squashed the gratuitous `pre-commit` commit, which I hadn't noticed before). I don't really care about the ordering of the merges. I can easily rebase the other branch when this is merged.

I'll continue the 100$q / fuller_name discussion on #7409.
